### PR TITLE
enhancment: optional file loader in `GoogleDriveLoader`

### LINKED ISSUE #5791: Enable the use of Unstructured loaders with Google drive loader
### Feature request

The goal of this issue is to enable the use of Unstructured loaders in conjunction with the Google drive loader.

### Motivation

This would enable the use of the GoogleDriveLoader with document types other than the standard Google Drive documents/spreadsheets/presentations.

### Your contribution

I can create a PR for this issue as soon as I have bandwidth. If another community member picks this up first, I'd be happy to review.

### PR DESCRIPTION
**Pull request by MthwRobinson**
### Summary

Gives users the ability to specify a `file_loader_cls` for processing files in Google Drive that are not Google Documents or Google Sheets.

Fixes #5791. See also [this Twitter thread](https://twitter.com/klaudioz/status/1666124861482184708?s=46&t=UImYT1AvX587tYTQ2i1sIQ) where a user requested this capability.

### Testing

For individual files (use a file id from your own GDrive):

```python
from langchain.document_loaders import GoogleDriveLoader
from langchain.document_loaders import UnstructuredFileIOLoader

file_id="1x9WBtFPWMEAdjcJzPScRsjpjQvpSo_kz"
loader = GoogleDriveLoader(
    file_ids=[file_id],
    file_loader_cls=UnstructuredFileIOLoader,
    file_loader_kwargs={"mode": "elements"}
)
loader.load()
```

For a folder:

```python
from langchain.document_loaders import GoogleDriveLoader
from langchain.document_loaders import UnstructuredFileIOLoader

folder_id="1asMOHY1BqBS84JcRbOag5LOJac74gpmD"
loader = GoogleDriveLoader(
    folder_id=folder_id,
    file_loader_cls=UnstructuredFileIOLoader,
    file_loader_kwargs={"mode": "elements"}
)
loader.load()
```

#### Who can review?

@hwchase17
@eyurtsev

### CODE REVIEW

File: langchain/document_loaders/googledrive.py [hunk 1]
```diff
@@ -279,23 +286,32 @@ def _load_file_from_id(self, id: str) -> List[Document]:
...
+            from PyPDF2 import PdfReader
+
+            content = fh.getvalue()
+            pdf_reader = PdfReader(BytesIO(content))
+
+            return [
+                Document(
+                    page_content=page.extract_text(),
+                    metadata={
+                        "source": f"https://drive.google.com/file/d/{id}/view",
```
Comment (hwchase17) [inline langchain/document_loaders/googledrive.py:308]: @MthwRobinson do you not want this metadata present as well?

wouldnt it make sense for the different loaders to have at least some common metadata

Comment (MthwRobinson) [reply langchain/document_loaders/googledrive.py:308]: Yeah that definitely makes sense, just missed that in there. I'll get that updated.

Comment (MthwRobinson) [reply langchain/document_loaders/googledrive.py:308]: @hwchase17 - Anything else you wanted to add to this one?

**Comment by RobSpectre** [comment]
Thanks for doing this @MthwRobinson - super helpful.

**Pull request merged**